.devcontainer/devcontainer.json

```diff
@@ -2,11 +2,23 @@
   "name": "Home Assistant Dev",
   "context": "..",
   "dockerFile": "../Dockerfile.dev",
-  "postCreateCommand": "script/setup",
+  "features": {
+    "ghcr.io/devcontainers/features/common-utils:1": {}
+  },
+  "containerUser": "vscode",
+  "postCreateCommand": "script/setup && pre-commit install --install-hooks",
```

Comment:
Installing the hooks separately should not be needed.

Reply:
`postCreate` happens in background after the container is already open. Installing the pre-commit hooks at this stage prevents the user from having to wait a long time when they need to run their first `git commit`.

Reply:
Let's adjust it, so it is applied earlier; since we pre-built dev containers, that would help with that too.

Reply:
No, there are other stages beside postCreate ;)

Reply:
Ok, I changed to `onCreate` it indeed should help with pre-building, but for the regular users this will slow down the startup of the container by several seconds.